sql/catalyst/src/main/java/org/apache/spark/sql/connector/write/RequiresDistributionAndOrdering.java

```diff
@@ -42,6 +42,19 @@
    */
   Distribution requiredDistribution();
 
+  /**
+   * Returns the number of partitions required by this write.
+   * <p>
+   * Implementations may override this to require a specific number of input partitions.
+   * <p>
+   * Note that Spark doesn't support the number of partitions on {@link UnspecifiedDistribution},
+   * if requiredDistribution() returns {@link UnspecifiedDistribution}, the return value for this
+   * method will be ignored.
```

Comment:
Shall we ignore or fail? It's very easy to detect it before applying the requirement.

Reply:
Let's make it fail so that the intention isn't silently ignored. Good point.